Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1897,18 +1901,26 @@ private void AddStaticFileDependencyToFileAccessManifest(FileArtifact dependency
             // Note that they would perhaps fail otherwise (simplifies the observed access checks in the first place).
 
             var path = dependency.Path;
+
+            bool pathIsUnderSharedOpaque = TryGetContainingSharedOpaqueRoot(path, getOutmostRoot: true, out var sharedOpaqueRoot);
+
             m_fileAccessManifest.AddPath(
                 path,
                 values: FileAccessPolicy.AllowRead | FileAccessPolicy.AllowReadIfNonexistent,
-                mask: m_excludeReportAccessMask & ~FileAccessPolicy.AllowRealInputTimestamps); // Make sure we fake the input timestamp
+                // Make sure we fake the input timestamp
+                // The file dependency may be under the cone of a shared opaque, which will give write access

```

Comment:
Doesn't the same happen when such a file is under the cone of an **exclusive** opaque directory?

Reply:
Inputs cannot be part of exclusive opaques, those are blocked by construction

---
In reply to: [284863123](https://github.com/microsoft/BuildXL/pull/316#discussion_r284863123) [](ancestors = 284863123)